### PR TITLE
Added a config script to auto detect the Python version to use.

### PR DESCRIPTION
**Pull request by hernando**
Fixed FindNumPy script to work with Python3.

### CODE REVIEW

File: ChoosePython.cmake [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+set_property(CACHE USE_PYTHON_VERSION PROPERTY STRINGS 2 3 auto)
+
+if(NOT PYTHONINTERP_FOUND AND ${USE_PYTHON_VERSION} STREQUAL auto)
+  find_package(PythonInterp 3 QUIET)
+  if(CHOOSE_PYTHON_REQUIRE_BOOST)
+    find_package(Boost COMPONENTS python3 QUIET)
+  endif()
+
+  if(PYTHONINTERP_FOUND AND (NOT CHOOSE_PYTHON_REQUIRE_BOOST OR Boost_FOUND))
+    set(USE_PYTHON_VERSION 3)
```
Comment (eile) [inline ChoosePython.cmake]: iirc this does not overwrite the cache value, are you sure it works?

File: ChoosePython.cmake [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    message(STATUS "Python 3 choosen automatically")
```
Comment (eile) [inline ChoosePython.cmake]: typo 'chosen'

**Comment by eile** [comment]
+1

File: ChoosePython.cmake [hunk 3]
```diff
@@ -0,0 +1,86 @@
+##
+## Molio
```
Comment (tribal-tec) [inline ChoosePython.cmake]: Not anymore :)

File: ChoosePython.cmake [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  endif()
+else()
+  set(PythonLibs_FIND_VERSION 2)
+  set(PythonInterp_FIND_VERSION 2)
+endif()
+
+execute_process(COMMAND
+  ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(1,0,prefix=''))"
+  OUTPUT_VARIABLE PYTHON_LIBRARY_SUFFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+message(STATUS "Python package suffix path: ${PYTHON_LIBRARY_SUFFIX}")
```
Comment (tribal-tec) [inline ChoosePython.cmake]: Why is it suffix? iirc, it's the path pointing to the python libs (/usr/lib/python-2..7 for instance), hence it's the prefix, no?

File: FindBoostConfig.cmake [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Boost settings to ensure that the Boost finder works
```
Comment (tribal-tec) [inline FindBoostConfig.cmake]: Proper copyright header?

File: ChoosePython.cmake [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  else()
+    set(USE_PYTHON_VERSION 2)
+    message(STATUS "Python 2 choosen automatically")
+  endif()
+endif()
+
+if(${USE_PYTHON_VERSION} STREQUAL 3)
+  set(${PROJECT_NAME}_USE_PYTHON3 ON)
+  # Enforcing a Python version to be searched by scripts included by
+  # Common.cmake that search for Python (e.g. ccplint)
```
Comment (tribal-tec) [inline ChoosePython.cmake]: typo ccplint

File: ChoosePython.cmake [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  set(PYTHON_ADDITIONAL_VERSIONS 3.4 3.3 3.2)
+  set(Python_ADDITIONAL_VERSIONS 3.4 3.3 3.2)
+  set(PythonLibs_FIND_VERSION 3)
+  set(PythonInterp_FIND_VERSION 3)
+  add_definitions(-D${PROJECT_NAME}_USE_PYTHON3=1)
+  set(${PROJECT_NAME}_USE_BOOST_PYTHON_VERSION 3)
+  # This shouldn't be necessary but the helps detecting the Python libs
```
Comment (tribal-tec) [inline ChoosePython.cmake]: grammar: ..but the helps..

File: ChoosePython.cmake (hunk 1 above)
Comment (hernando) [inline ChoosePython.cmake]: It doesn't, but It's not meant to rewrite the cache value (actually I don't care whether it does or not), but to pass the detected version downstream.

File: ChoosePython.cmake (hunk 4 above)
Comment (hernando) [inline ChoosePython.cmake]: It's a suffix of the prefix path (append to the install prefix), I thought that prefix was a bit misleading.

Comment (hernando) [inline ChoosePython.cmake]: PackageConfig.cmake:108:

    PATHS \${${PROJECT_NAME}_PREFIX_DIR} PATH_SUFFIXES lib ${PYTHON_LIBRARY_SUFFIX})

Having PYTHON_LIBRARY_PREFIX there looked weird to me.

Comment (tribal-tec) [inline ChoosePython.cmake]: Ok then

File: FindBoostConfig.cmake [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2012-2015, Stefan Eilemann <eile@eyescale.ch>
+#               2013-2015, Daniel Nachbhar <dnachbaur@epfl.ch>
```
Comment (tribal-tec) [inline FindBoostConfig.cmake]: nope, that's not my name and email :)

**Comment by tribal-tec** [comment]
+1 if my name is restored

**Pull request merged**